port/unix/omrsysinfo.c

```diff
@@ -3568,13 +3582,28 @@ omrsysinfo_startup(struct OMRPortLibrary *portLibrary)
 	 */
 	if (0 == attachedPortLibraries) {
 		if (omrthread_monitor_init_with_name(&cgroupEntryListMonitor, 0, "cgroup entry list monitor")) {
-			return OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			rc = OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			goto _end;
 		}
 	}
 	attachedPortLibraries += 1;
-	isRunningInContainer(portLibrary, &PPG_isRunningInContainer);
+
+	if (isCgroupV1Available(portLibrary)) {
+		PPG_sysinfoControlFlags |= OMR_CGROUP_V1_AVAILABLE;
+	} else if (isCgroupV2Available()) {
+		PPG_sysinfoControlFlags |= OMR_CGROUP_V2_AVAILABLE;
+	}
+	if (0 == isRunningInContainer(portLibrary, &runningInContainer)) {
+		if (runningInContainer) {
+			PPG_sysinfoControlFlags |= OMR_RUNNING_IN_CONTAINER;
+		}
+	} else {
+		rc = OMRPORT_ERROR_STARTUP_SYSINFO_RUNNING_IN_CONTAINER;
+	}
+
+_end:
 #endif /* defined(LINUX) */
-	return 0;
+	return rc;
 
 #if (defined(S390) || defined(J9ZOS390))
 	PPG_stfleCache.lastDoubleWord = -1;
```

Comment:
This dead code was added in #4662: it should either be removed or moved to where it belongs.